pkg/postgres/schema/all.go

```diff
@@ -21,17 +21,22 @@ var (
 )
 
 type registeredTable struct {
-	Schema     *walker.Schema
-	CreateStmt *postgres.CreateStmts
+	Schema          *walker.Schema
+	CreateStmt      *postgres.CreateStmts
+	FeatureFlagFunc func() bool
 }
 
 // RegisterTable maps a table to an object type for the purposes of metrics gathering
-func RegisterTable(schema *walker.Schema, stmt *postgres.CreateStmts) {
+func RegisterTable(schema *walker.Schema, stmt *postgres.CreateStmts, featureFlagFuncs ...func() bool) {
```

Comment:
Interesting. So we could use this to register new schemas within the code, but would there be a corresponding migration script?

I was looking through the migration scripts/documentation earlier and noticed they were applied sequentially, but this doesn't appear to be related to those migrations.

Reply:
Would we keep this opt-in registration specific to new tables? For example, would this evolve to support optional schemas that aren't table additions?

Reply:
So the schemas are created and migrated at the end of the migrator. In this case, this schema would not be created

Reply:
Maybe the migration conversation could be taken off line, but how would this work for a migration?  The fact they run in sequence would make it difficult to put a migration behind a gate OR is the idea we are only going to write the migration at the end when we are ready to remove the gate?

Reply:
This is a good question, but we'll need to figure out if we can effectively flag migrations